### ISSUE TITLE
memory corruption via operator =

### ISSUE THREAD

**Issue by GoogleCodeExporter** [byzhang/rapidjson#22]
```
What steps will reproduce the problem?
1. compile and run the simple attached program
2.
3.

What is the expected output? What do you see instead?
No output, just successful completion.  Memory corruption on free().

What version of the product are you using? On what operating system?
Recent, Linux.

Please provide any additional information below.

I think perhaps operator= should be private?
```

Original issue reported on code.google.com by `brian.budge@gmail.com` on 25 Apr 2012 at 7:32

Attachments:
- [foo.cpp](https://storage.googleapis.com/google-code-attachments/rapidjson/issue-22/comment-0/foo.cpp)

**Comment by GoogleCodeExporter** [comment]
Original comment by `milo...@gmail.com` on 26 Apr 2012 at 3:02
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r74.
```

Original comment by `milo...@gmail.com` on 14 Nov 2012 at 2:44
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r75.
```

Original comment by `milo...@gmail.com` on 14 Nov 2012 at 2:47

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 9fb77b114b74.
```

Original comment by `milo...@gmail.com` on 6 Jun 2014 at 6:04
